src/api/argo-templates/base/sat-general-template-iuf.argo.yaml

```diff
@@ -0,0 +1,146 @@
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+apiVersion: argoproj.io/v1alpha1
+kind: WorkflowTemplate
+metadata:
+  name: sat-general-template
+spec:
+  entrypoint: sat-wrapper
+  templates:
+    ### Templates ###
+    ## sat-wrapper ##
+    - name: sat-wrapper
+      nodeSelector:
+        kubernetes.io/hostname: ncn-m001
+      tolerations:
+        - key: node-role.kubernetes.io/master
+          effect: NoSchedule
+      metadata:
+        annotations:
+          sidecar.istio.io/inject: "false"
+      inputs:
+        parameters:
+          - name: ca_certs_bundle
+            value: /etc/ssl/certs/ca-certificates.crt
+          - name: auth_token
+          - name: media_dir
+            description: >
+              The path to the IUF media directory that will be mounted in the sat container.
+              This is required to allow the sat commands to access any necessary files, e.g.
+              sat bootprep input files. This will be mounted at the same path within the sat
+              container.
+          - name: script_stdout_file
+            default: "/tmp/script_stdout_file"
+          - name: script_content
+            description: >
+              The script that will be executed after the sat CLI is configured. The stdout
+              of this script will be captured and returned as an output parameter from this
+              template.
+      outputs:
+        parameters:
+          - name: script_stdout
+            valueFrom:
+              path: "{{inputs.parameters.script_stdout_file}}"
+      script:
+        image: artifactory.algol60.net/sat-docker/stable/cray-sat:3.20.0
+        command: [sh]
+        source: |
+          #!/bin/sh
+          ts="$(mktemp)"
+          cat <<'SAT_GENERAL_EOF' > "${ts}"
+          function setup() {
+            set -e
+            sat init
+
+            # Write the auth token parameter to the file
+            JSON_FMT='{"access_token":"%s"}\n'
+            printf "$JSON_FMT" "{{inputs.parameters.auth_token}}" > /root/.config/sat/token.json
+            echo -n $(tr -d "\n\r" < /root/.config/sat/token.json) > /root/.config/sat/token.json
+
+            # Set up the SAT config to use the token file
+            sed -i -e 's|.*token_file.*|token_file = "/root/.config/sat/token.json"|' /root/.config/sat/sat.toml
+
+            # TODO: Fix sat bootprep code to load kube config from cluster
```

Comment:
I filed [CRAYSAT-1656](https://jira-pro.its.hpecorp.net:8443/browse/CRAYSAT-1656) for this, and it might get done in time for Alice. If it does, we can remove this workaround then.